Feat/embed

AlyshaNolan opened the pull request:
# Context

[Link to Notion ticket](https://www.notion.so/cloudcannon/Embed-component-1a85a7ab79e3451f97f8693b2270364e?pvs=4)
and for the notification banner - [Link to Notion ticket](https://www.notion.so/cloudcannon/Notification-e4af1d0807c542f397f9cfb6ae73a931?pvs=4)

# Additional information

The embed will not work in live editing if it has a script (hence why I developed the note here)
Have tested with:
- Lottie File
- PDF
- Google docs
- Hubspot form

# Testing (tester)

The reviewer should check on this branch:

- The code
- The component in Bookshop browser
- The site in CloudCannon ([link here](https://app.cloudcannon.com/42158/editor#sites/118597/collections/pages/))

# Before merge (PR owner)

- [ ] Delete the test site in CloudCannon
- [ ] For "generic" components **only**: remove `- content_blocks` from `structures` in the YAML file (this was added for testing).

AlyshaNolan committed:
Feat: create notification and embed components

AlyshaNolan committed:
Fix: key

AlyshaNolan committed:
Feat: seperate the embed to its own component with more options

AlyshaNolan committed:
Merge branch 'main' into feat/embed

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon commented:
What _should_ this component support? Tried with YouTube embed - got a lot of odd behaviour... 
- if this should be a generic 'everything' embed, then needs more work/testing on different types etc...
- if it is specifically _not_ for things like YouTube videos etc... then it probably needs some text to that effect, maybe a list of supported providers etc...
- which then begs the question, should there be a 'media' embed component for embedding things like YouTube videos etc?

adon-cloudcannon commented:
Also - tested with a Google Doc - did not seem to resize correctly using the ratio option thingee... could see the container resizing, but the actual embedded content remains small... again, not sure what the intended behaviour is? If this is something that can't be controlled, should the option be there?

adon-cloudcannon commented:
Question: should the custom embed be on the component by default? I realize that they will need to add it but what is the point of the embed component _without_ it, you know?

- also: does the embed code block need a description or comment? I am used to this look as a developer, but this is for editors... perhaps they will not realize this is a code block to paste their code into? Maybe even some basic instructions along the lines of "Go to the service you want to embed, copy the code as per their instructions, and paste the code in here"... idk...

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon commented:
There seems to be a weird caching bug (at least on Chrome) with Google docs - this could be a browser thing, a GD thing, or a CloudCannon thing... do we care?
- Updating the Google Doc does not update in the published CC site (or live editor)... hard refresh with cache clearing seems to fix? (I don't know what default GD behaviour for embeds is meant to be - it _sounds like_ it should update automatically? Investigate, or don't care?

adon-cloudcannon commented:
Can't seem to remove the custom width px after being set - deleting the number then just defaults it to 0, and hides the embedded content.
- setting to -1 makes the default 100% width work again... but I assume this is just because I've broken the px and it's defaulting back to 100% - this could be a hacky workaround to leave in there for the user, but then there needs to be some text to this effect (put -1 to make it default back to 100%.... that sort of instruction?)

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

AlyshaNolan committed:
Fix: show/hide the note better and add comments

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

Commit:
Reactor: removing the max-width and trying to center all embeds

- changed the .c-custom-embed to display:flex and justify-content:center
- removed the explicit width calculation part
- the idea is that the content will default to 100% width and if the user explicitly sets a different width in their embed code, that's on them

Commit:
Merge branch 'feat/embed' of github.com:CloudCannon/small-business-template-eleventy into feat/embed

Commit:
Chore: remove other references to max-width

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

AlyshaNolan committed:
Feat: define better add options

AlyshaNolan committed:
Updated 1 file via CloudCannon.

AlyshaNolan committed:
Feat: move ratio out of styles object

AlyshaNolan committed:
Chore: match component nesting

AlyshaNolan committed:
Fix: alignment

AlyshaNolan committed:
Updated 1 file via CloudCannon.

adon-cloudcannon committed:
Updated 1 file via CloudCannon.

Commit:
Fix: added conditional check for heading or text

- if neither is present, hide the component entirely - fixes extra gap bug
- brings it in line with other components doing the same thing

Commit:
Merge branch 'feat/embed' of github.com:CloudCannon/small-business-template-eleventy into feat/embed

adon-cloudcannon commented:
It's great! Alignment works well... I have:
- Added the same conditional check for heading OR description - if neither, hide component... same as in centred asset block (should probably do a pass at the end for all components?

Only other comment would be maybe we want different orientations for A4 and video in the dropdowns? But that's a thing we can always add to later... I reckon merge this in to get it off the tickets and we can do a 'nice to have' run at the end if we have time?

adon-cloudcannon reviewed (approved):
See comment :)

AlyshaNolan committed:
Chore: remove components from page
